i8_terminal/service_result/service_result.py

```diff
@@ -73,24 +75,39 @@ def _process_value(raw: Union[int, float], formatted: str, ci: ColumnInfo) -> st
     def to_plot(self, x: str, y: List[str], kind: str = "bar") -> Any:
         return self._to_plot(x, y, kind)
 
-    def _to_plot(self, x: str, y: List[str], kind: str = "bar") -> Any:
+    def _to_plot(self, cmd_context: Dict[str, Any], x: str, y: List[str], kind: str = "bar") -> None:
         df = self._df[[x] + y]
-
-        df_grouped = df.groupby(x)[y].mean()
-        ax = df_grouped.plot(kind=kind, figsize=(10, 6))
+        df_grouped = df.groupby(x)[y].mean().reset_index()
 
         cid = self._cols_context.get_col_info_dict()
         y_display_names = [cid[ci_name].display_name for ci_name in y]
         title = f"{concat_and(y_display_names)} by {cid[x].display_name}"
-
-        ax.set_title(title)
-        ax.set_xlabel(cid[x].display_name)
-        ax.legend(y_display_names)
-
-        plt.xticks(rotation=45, ha="right")
-        plt.show()
-
-        return plt
+        cmd_context["plot_title"] = title
+
+        fig = None
+        if kind == "bar":
+            fig = px.bar(
+                df_grouped,
+                hover_data=[],
+                x=x,
+                y=y,
+                barmode="group",
+                title=title,
+                labels={"value": "Metric Value", x: "Period"},
+            )
+        if not fig:
+            return
+
+        fig.update_traces(width=0.3, hovertemplate="%{y}%{_xother}")
+        fig.update_layout(
+            **get_plot_default_layout(),
+            bargap=0.4,
+            legend_title_text=None,
+            xaxis_title=None,
+            margin=dict(b=15, l=70, r=20),
+        )
+
+        serve_plot(fig, cmd_context)
```

Comment:
We don't need to serve the plot when its used in the notebook. The plot should be drawed in the notebook